js/views/userPageVw.js

```diff
@@ -1444,6 +1467,70 @@ module.exports = baseVw.extend({
 
   },
 
+  addModeratorClick: function(e){
+    var $targ = $(e.target).closest('.js-addmoderator'),
+        self = this,
+        modList = {};
+
+    $targ.addClass('loading');
+
+    modList.moderators = this.model.get('user').moderator_guids;
+    modList.moderators.push(this.pageID);
+
+    saveToAPI('', this.model.get('user'), this.model.get('user').serverUrl + "settings",
+      function(){
+        // confirmed
+        self.options.userModel.fetch({
+          success: function(model, response) {
+            if (self.isRemoved()) return;
+            var user = self.model.get('user');
+            user.moderators = model.get('moderators');
+            user.moderator_guids = model.get('moderator_guids');
```

Comment:
This is still a bit over-complex.

The page model is just an object with two models inside it, user (userMd.js) and page (userProfile.js). It's done this way because the template needs access to data from both of them, but it makes things a little klunky.

The fetch is updating the user model, all you need to do is replace the existing user object with this model after changing it to json.

Something like self.model.set('user', model.toJSON()) <-- untested pseudocode